### PR TITLE
Feature/sonarlint/data/link

### PR DESCRIPTION
**Pull request by PedroErnst**
Oversized lines
Missing braces for control statements
Indentation 4 spaces
Oversized lines
Formatting
Explicitly state variables and methods as public or private
Remove spare empty lines
Remove dead code
Code smell

## Motivation and Context

Make the file easiert to understand and maintain

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: data/Link.php [hunk 1]
```diff
@@ -1,9 +1,12 @@
 <?php
-if(!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
+
```
Comment (daniel-samson) [inline data/Link.php]: Can we please update the license file?

File: data/Link.php [hunk 2]
```diff
@@ -39,682 +42,733 @@
...
+ * be used to manipulate relationship between object instances.
+ * Portions created by SugarCRM are Copyright (C) SugarCRM, Inc.
+ * All Rights Reserved.
+ * Contributor(s): ______________________________________..
+ ********************************************************************************/
+
+class Link
+{
+    /* Private variables.*/
+    private $_relationship_name; //relationship this attribute is tied to.
```
Comment (daniel-samson) [inline data/Link.php]: Can we also add the PHP DOC for each private variable?

File: data/Link.php [hunk 3]
```diff
@@ -39,682 +42,733 @@
...
+    private $_duplicate_where;
+
+    /* Parameters:
+    * 		$_rel_name: use this relationship key.
+    * 		$_bean: reference of the bean that instantiated this class.
+    * 		$_fieldDef: vardef entry for the field.
+    * 		$_table_name: optional, fetch from the bean's table name property.
+    * 		$_key_name: optional, name of the primary key column for _table_name
+    */
+    public function __construct($_rel_name, &$_bean, $fieldDef, $_table_name = '', $_key_name = '')
```
Comment (daniel-samson) [inline data/Link.php]: Please update the PHP Doc so that it shows the  `@param type $param_name`

File: data/Link.php [hunk 4]
```diff
@@ -39,682 +42,733 @@
...
+    /* This method will return the following based on cardinality of the relationship.
+    *  # one-to-many, many-to-many: empty array if not data is found else array of keys.
+    *  # if many-to-many and $role set to true : empty array if not data is found else
+    *  array of array which contain id+other fields.
+    *  # many-to-one, one-to-one: null if no linked data is found, else key value.
+    *
+    * For a self referencing relationship the function will behave as if the user is trying
+    * to access the child records. To get to the parent records use the getParent() method.
+    */
+    public function get($role = false)
```
Comment (daniel-samson) [inline data/Link.php]: Please update the PHP Doc To reflect the @params

File: data/Link.php [hunk 5]
```diff
@@ -39,682 +42,733 @@
...
-
-		$GLOBALS['log']->debug("Link Constructor, _bean_table_name: ".$this->_bean_table_name);
-		$GLOBALS['log']->debug("Link Constructor, _bean_key_name: ".$this->_bean_key_name);
-		if (!empty($this->_relationship->id)) $GLOBALS['log']->debug("Link Constructor, relationship record found.");
-		else $GLOBALS['log']->debug("Link Constructor, No relationship record.") ;
-
-	}
-
-    function loadedSuccesfully() {
+    public function loadedSuccesfully()
```
Comment (daniel-samson) [inline data/Link.php]: Missing PHPDOC

**Comment by PedroErnst** [comment]
@daniel-samson Updated the PR

**Pull request closed without merging**